File: TrabajoI.C/JS/claseAbstracta/ClaseD.js
class D extends Figura {
    constructor(obj) {
        super();

        if (obj instanceof D) {
            // Constructor copia
            this.data = obj.data;
            console.log(`D copia → data: ${this.data}`);
        } else {
            this.data = obj || 0;
            console.log(`D normal → data: ${this.data}`);
        }
    }

    dibujar() {
        console.log("Dibujando D");
    }
}

File: TrabajoI.C/C#/claseAbstracta/ClaseD.cs
public class D : Figura
{
    public int Numero { get; set; }

    public D(int numero)      // Constructor normal
    {
        Numero = numero;
        Console.WriteLine($"Constructor normal en D: {numero}");
    }

    public D(D original)      // Constructor copia
    {
        Numero = original.Numero;
        Console.WriteLine($"Constructor copia en D: {Numero}");
    }

    public override void Dibujar()
    {
        Console.WriteLine("Dibujando D");
    }
}
